### DISCUSSION TITLE
Multi-lingual sourcing

### DISCUSSION THREAD

**Discussion by m99coder** [gatsbyjs/gatsby#32094] [Ideas / Feature Requests]
First things first: Thanks for the renewed Shopify source plugin.

Since you are now using Shopify’s Bulk Operations API, I wonder if there will be support for different languages as there is for quite some time in the Storefront API with the [translation header](https://shopify.dev/tutorials/manage-app-translations-with-admin-api#storefront-api-translation-header)?

From GatsbyConf 2021, I learned there will be multi-lingual support. Is there any timeline worth sharing or ready to do so? Any proof-of-concept to get an idea of how you will integrate this very important feature. More and more e-commerce sites are at least bi-lingual.

All the best,
Marco

**Comment by sslotsky** [comment]
Thank you for the issue. We do indeed plan to support this. While our strategy hasn't been officially decided, the easiest way to support it is probably by [retrieving translations from resources where applicable](https://shopify.dev/tutorials/manage-app-translations-with-admin-api#retrieve-translations-from-a-resource-object). Plugin users would specify which languages they need in the `pluginOptions` and a translatable resource would then have translation entries for each provided language.

Does that sound like something that could work for your case?

**Comment by m99coder** [comment]
Could you maybe elaborate a little bit on the `pluginOptions` part? It would help to see a tiny example (even if only pseudo-code). If there is a way to retrieve the `translations` fields for specific resource fields (as supported by Shopify), that would absolutely work in my case. As far as I know the Storefront API doesn’t yet support querying for `translations`. I could only make it work with the Admin API. Not sure if the Batch API supports them.

**Comment by sslotsky** [comment]
You would for example pass something like this into plugin options:

```
languages: ['fr']
```

Then perhaps we add something like this to our bulk query:

```gql
        fr_translations: translations(locale: "fr") {
          key,
          locale,
          value
        }
```

And now your resource has a field `fr_translations` with that translation data.

**Reply by ilbertt** [reply]
I guess this feature is not yet available right? Is there a workaround?

**Reply by ucimo** [reply]
I can't beleive this feature is so hard to implement with Gatsby and Shopify. Soooo much glofication on the internet for a software that dosn't offer a straight forward way to use a translations... smh

**Comment by m99coder** [comment]
@sslotsky Perfectly fine 👍

**Comment by DennisKraaijeveld** [comment]
Any progress on this? I thought this would be available after all this time

**Comment by RectoVersoDev** [comment]
https://www.gatsbyjs.com/resources/webinars/high-performance-localization-storyblok/

**Reply by AlexanderProd** [reply]
This is not about Shopify though, is it?

**Reply by RectoVersoDev** [reply]
I don't think so but there might be some interesting information 🤞

**Comment by Karlos137** [comment]
Hello @sslotsky , any progress on this? Is there an alternative that we can use meanwhile? Thank you.
